src/lab3/structures/traceable-group/TraceableGroup.ts

```diff
@@ -21,3 +22,24 @@ export default class TraceableGroup<T extends Traceable> implements Traceable {
     return closestHit;
   }
 }
+
+export class SceneObjectGroup<T extends SceneObject>
+  extends TraceableGroup<T>
+  implements SceneObject
+{
+  translate(x: number, y: number, z: number): void {
+    for (const traceableObject of this.traceableObjects) {
+      traceableObject.translate(x, y, z);
+    }
+  }
+  rotate(angleX: number, angleY: number, angleZ: number): void {
+    for (const traceableObject of this.traceableObjects) {
+      traceableObject.rotate(angleX, angleY, angleZ);
+    }
+  }
+  scale(x: number, y: number, z: number): void {
+    for (const traceableObject of this.traceableObjects) {
+      traceableObject.scale(x, y, z);
+    }
+  }
```

Comment:
This logic should be implemented in `TraceableGroup`

Reply:
TraceableGroup is not responsible for transforms. It only exposes Traceable interface. SceneObjectGroup extends it and adds Transformable interface to it and implements this. I'd like to keep the original class intact